### PR TITLE
feat: attach custome artifacts

### PR DESCRIPTION
**Pull request by tianfeng92**
## Description

Introduced `sauceAttachments` to upload artifact.

Check example in https://github.com/saucelabs/testcafe-reporter/actions/runs/11825219821/job/32948498364?pr=63

### CODE REVIEW

**Review by alexplischke — COMMENTED** [review]
How does this solution work with TestCafe [running concurrently](https://testcafe.io/documentation/403626/guides/intermediate-guides/run-tests-concurrently)?

**Comment by alexplischke** [comment]
@tianfeng92 Is https://testcafe.io/documentation/404350/reference/test-api/testcontroller/report something that could be utilized to pass additional files or file information to the reporter instead?

**Comment by tianfeng92** [comment]
> @tianfeng92 Is https://testcafe.io/documentation/404350/reference/test-api/testcontroller/report something that could be utilized to pass additional files or file information to the reporter instead?

This could work. 👍  I've tried to report the dir name
```
t.report(
        {'artifactUploadDir': 'assets/random_unique_dir'},
);
```
and the concurrent reported data `testRunInfo.reportData` is like
```
{
  "Gcd5EKIkZ": [
    {
      "artifactUploadDir": "assets/random_unique_dir"
    }
  ],
  "3jS4iw7xM": [
    {
      "artifactUploadDir": "assets/random_unique_dir"
    }
  ]
}
```

`Gcd5EKIkZ` and `3jS4iw7xM` are testRunIds and I think if it's setting like this, the concurrent scenario can be handled.

File: src/json-reporter.js [hunk 1]
```diff
@@ -211,6 +219,24 @@ function reporterFactory() {
...
+      for (const attachment of attachments) {
+        try {
+          if (fs.existsSync(attachment) && fs.statSync(attachment).isFile()) {
+            assets.push({
+              name: path.basename(attachment),
+              localPath: attachment,
+            });
+          }
+        } catch (error) {
+          console.warn(`Skipping upload ${attachment}: ${error.message}`);
```
Comment (alexplischke) [inline src/json-reporter.js:234]:
```suggestion
          console.warn(`Skipping upload of ${attachment}: ${error.message}`);
```

**Review by alexplischke — APPROVED** [review]

File: src/json-reporter.js [hunk 2]
```diff
@@ -103,8 +104,14 @@ function reporterFactory() {
         const videoAssets = testRunInfo.videos
           .filter(idFilter)
           .map((v) => this._getAsset(v.videoPath, testName));
-        const testStartTime = this.startTimes.get(testRunInfo.testId);
 
+        const sauceAttachments =
+          testRunInfo.reportData[browser.testRunId]?.find(
+            (item) => item.sauceAttachments,
```
Comment (mhan83) [inline src/json-reporter.js:110]: `t.report` takes args of type `...args: any[]` so I'm wondering if we want to at least have null(ish) check on `item` in this find predicate. For example, I could see some doing something silly like 

```ts
t.report(variableThatIsUnexpectedlyNull, { sauceAttachments: [] })
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -34,3 +34,34 @@ jobs:
...
+
+      - name: Test
+        working-directory: tests/integration
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+        run: |
+          npm ci
+          npm link testcafe-reporter-saucelabs
+          node testcafe-integration-test.js
```
Comment (mhan83) [inline .github/workflows/test.yml:67]: Out of curiosity, why did you wrap the execution in a node script like this? The "test" script doesn't have any assertions, it's just a wrapper for a shell command. I'd think this is sufficient:

```suggestion
          npx testcafe chrome:headless tests --reporter saucelabs
```

Comment (tianfeng92) [reply .github/workflows/test.yml:67]: Ah nothing specific, just trapped in create a `createTestCafe` runner so I created this script. I just realize I can do in this way 😄

**Review by mhan83 — APPROVED** [review]

**Pull request merged**